plugin.py

```diff
@@ -134,15 +150,12 @@ def _handle_config_commands_async(self, settings_change_count: int) -> None:
                 settings[key] = values
             else:
                 print("LSP-lua: unrecognized action:", action)
-        session.window.set_project_data(base)
-        if not session.window.project_file_name():
-            sublime.message_dialog(" ".join((
-                "The server settings have been applied in the Window,",
-                "but this Window is not backed by a .sublime-project.",
-                "Click on Project > Save Project As... to store the settings."
-            )))
-
-    def _get_server_settings(self, window: sublime.Window) -> Tuple[Optional[Dict[str, Any]], Optional[Dict[str, Any]]]:
+        _update_project_data(window, base)
```

Comment:
Maybe only do this when something has changed? This will at the very least restart the servers and maybe also affect the tree state in the sidebar (not sure).